warehouse/src/main/java/io/skai/warehouse/cache/CacheStore.java

```diff
@@ -0,0 +1,29 @@
+package io.skai.warehouse.cache;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class CacheStore<T> {
+
+    private final Cache<Long, T> cache;
+
+    public CacheStore(int expiryDuration, TimeUnit timeUnit) {
+        cache = CacheBuilder.newBuilder()
```

Comment:
Please use CacheLoader, it's more sophisticated tool, it use computeIfAbsent approach instead of manual insertion/retrieving management.